api/src/services/platform-service.ts

```diff
@@ -92,6 +99,196 @@ export class PlatformService extends DBService {
+        throw new HTTP400('no s3Key found');
+      }
+
+      publishIds.summaryInfo.summaryId = summaryData.id;
+
+      const s3File = await getFileFromS3(summaryData.key);
+      const artifactZip = new AdmZip();
+      artifactZip.addFile(summaryData.file_name, s3File.Body as Buffer);
+
+      const dataPackageId = emlService.packageId;
+      const artifact: IArtifact = {
+        dataPackageId,
+        archiveFile: {
+          data: artifactZip.toBuffer(),
+          fileName: `${summaryData.uuid}.zip`,
+          mimeType: 'application/zip'
+        },
+        metadata: {
+          file_name: summaryData.file_name,
+          file_size: String(s3File.ContentLength),
+          file_type: 'Summary',
+          title: summaryData.file_name,
+          description: summaryData.message
+        }
+      };
+
+      const { artifact_id } = await this._submitArtifactToBioHub(artifact);
+      publishIds.summaryInfo.artifactId = artifact_id;
+    }
+
+    /**
+     * Check for report, if report are present,
+     * then submit all reports to biohub as an artifact
+     */
+    if (data.reports.length !== 0) {
+      const reportIds = data.reports.map((report) => report.id);
+      await this.uploadSurveyReportAttachmentsToBioHub(emlService.packageId, projectId, reportIds);
+    }
+
+    /**
+     * Check for attachments, if attachments are present,
+     * then submit all attachments to biohub as an artifact
+     */
+    if (data.attachments.length !== 0) {
+      const attachmentIds = data.attachments.map((attachment) => attachment.id);
+      await this.uploadSurveyAttachmentsToBioHub(emlService.packageId, projectId, attachmentIds);
+    }
+
+    //Check security request and create DWCA file for submission
+    const securityRequest = await surveyService.getSurveyProprietorDataForSecurityRequest(surveyId);
+    const dwCADataset: IDwCADataset = {
+      archiveFile: {
+        data: dwcArchiveZip.toBuffer(),
+        fileName: `${emlService.packageId}.zip`,
+        mimeType: 'application/zip'
+      },
+      dataPackageId: emlService.packageId,
+      securityRequest
+    };
+
+    //Submit DWCA file to biohub
+    const queueResponse = await this._submitDwCADatasetToBioHubBackbone(dwCADataset);
+    publishIds.queueId = queueResponse.queue_id;
+
+    //Publish records to history
+    await this.publishHistory(projectId, surveyId, publishIds);
+
+    return { uuid: emlService.packageId };
+  }
+
+  /**
+   * Publishes the history of the submission to the database
+   *
+   * @param {number} projectId
+   * @param {number} surveyId
+   * @param {publishIds} publishIds
+   * @memberof PlatformService
+   */
+  async publishHistory(projectId: number, surveyId: number, publishIds: publishIds) {
+    const publishArray = [];
+    if (publishIds.queueId) {
+      publishArray.push(
+        this.publishService.insertProjectMetadataPublishRecord({
+          project_id: projectId,
+          queue_id: publishIds.queueId
+        })
+      );
```

Comment:
I don't think we want to mark the project metadata as having been published, after only publishing the survey dataset. 

Same goes in reverse when publishing the project dataset.

Reply:
I would just change this function to be survey specific, and make a separate one for project later.